src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +33,156 @@ ActorInit Dm_Hina_InitVars = {
+    this->unk160.z = this->actor.world.pos.z + 100.0f;
+    this->unk16C.x = this->actor.world.pos.x;
+    this->unk16C.y = this->actor.world.pos.y + this->unk154 * this->unk15C + 40.0f * this->unk15C;
+    this->unk16C.z = this->actor.world.pos.z;
+    Play_SetCameraAtEye(play, this->unk17A, &this->unk16C, &this->unk160);
+    Math_SmoothStepToF(&this->actor.world.pos.y, this->unk158 + 300.0f, 0.5f, 2.0f, 0.1f);
+    if (((this->unk158 + 240.0f) < this->actor.world.pos.y) && (this->unk17E != 1)) {
+        this->unk17E = 1;
+        Actor_PlaySfx(&this->actor, NA_SE_OC_WHITE_OUT_INTO_KYOJIN);
+    }
+    Actor_PlaySfx(&this->actor, NA_SE_EV_MASK_RISING - SFX_FLAG);
+}
+
+void func_80A1F75C(DmHina* this, PlayState* play) {
+    s32 i;
+
+    switch (this->unk17E) {
+        case 0:
+            this->unk17F = 0;
+            this->unk14C = 0.0f;
+            break;
+
+        case 1:
+            Math_SmoothStepToF(&this->unk14C, 1.0f, 0.4f, 0.05f, 0.001f);
+            this->unk17F = this->unk14C * 255.0f;
+            this->unk150 = Math_SinS(play->state.frames * 0x1F40);
+            for (i = 0; i < 3; i++) {
+                play->envCtx.lightSettings.ambientColor[i] = play->envCtx.lightSettings.fogColor[i] =
+                    play->envCtx.lightSettings.diffuseColor1[i] = -255.0f * this->unk14C;
+            }
+            play->envCtx.lightSettings.fogNear = -500.0f * this->unk14C;
+            if (play->envCtx.lightSettings.fogNear < -300) {
+                play->roomCtx.curRoom.segment = NULL;
+            }
+            break;
+    }
+}
+
+void DmHina_Update(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->actionFunc(this, play);
+    func_80A1F75C(this, play);
+}
+
+void func_80A1F9AC(DmHina* this, PlayState* play) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+
+    if (this->unk17E != 0) {
+        OPEN_DISPS(gfxCtx);
+
+        Gfx_SetupDL25_Xlu(play->state.gfxCtx);
+
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, (u8)(this->unk14C * 100.0f) + 155, this->unk17F);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, (u8)(this->unk150 * 100.0f) + 50, 0);
+        gSPDisplayList(POLY_XLU_DISP++, gLightOrbMaterial1DL);
+
+        Matrix_Translate(this->actor.world.pos.x,
+                         this->actor.world.pos.y + (this->unk154 * this->unk15C) + (40.0f * this->unk15C),
+                         this->actor.world.pos.z, MTXMODE_NEW);
+        Matrix_ReplaceRotation(&play->billboardMtxF);
+        Matrix_Scale(this->unk14C * 20.0f, this->unk14C * 20.0f, this->unk14C * 20.0f, MTXMODE_APPLY);
+        Matrix_RotateZF(Rand_ZeroFloat(2 * M_PI), MTXMODE_APPLY);
+
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, gLightOrbModelDL);
+
+        CLOSE_DISPS(gfxCtx);
+    }
+}
+
+void DmHina_Draw(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    if (this->unk180 != 0) {
```

Comment:
```suggestion
    if (this->unk180) {
```
`unk180` looks bool, could make sure it's always set to `true` and `false`. Could also name this variable `isDrawn` if you'd like